### PR TITLE
fix collisions with projected columns and order by columns

### PR DESCRIPTION
**Pull request by WaVEV**

### CODE REVIEW

**Comment by timgraham** [comment]
I made a couple edits on the Django fork (see last commit there), so a few more aggregation_regress tests cases can run. Do you want to check the "To triage:" list under expected failures before merging (rather than adding more details in a follow-up commit)?

**Comment by WaVEV** [comment]
> I made a couple edits on the Django fork (see last commit there), so a few more aggregation_regress tests cases can run. Do you want to check the "To triage:" list under expected failures before merging (rather than adding more details in a follow-up commit)?

yes, what is "the triage"? where can I find it?

File: django_mongodb/features.py [hunk 1]
```diff
@@ -61,6 +61,17 @@ class DatabaseFeatures(BaseDatabaseFeatures):
...
         "aggregation.tests.AggregateTestCase.test_reverse_fkey_annotate",
+        "aggregation_regress.tests.AggregationTests.test_annotation_disjunction",
+        "aggregation_regress.tests.AggregationTests.test_decimal_aggregate_annotation_filter",
+        # QuerySet.extra(select=...) should raise NotSupportedError instead of:
+        # 'RawSQL' object has no attribute 'as_mql'.
+        "aggregation_regress.tests.AggregationTests.test_annotate_with_extra",
+        "aggregation_regress.tests.AggregationTests.test_annotation",
+        "aggregation_regress.tests.AggregationTests.test_more_more3",
+        "aggregation_regress.tests.AggregationTests.test_more_more_more3",
+        # To triage:
```
Comment (timgraham) [inline django_mongodb/features.py:72]: Here!

Comment (WaVEV) [reply django_mongodb/features.py:72]: Great tests, in the first one we have two issues: 

1. It is related with joins.
2. the way that we are storing object_ids in Django models. we cannot store a char[24].

Note that : `collection.find({'_id': '66c42726db971422470255ed'})` it won't match the thing that have the id `ObjectId('66c42726db971422470255ed')`.

Comment (WaVEV) [reply django_mongodb/features.py:70]: the second test is fixed, just another collision. Fixed in the last commit and test removed from features.py

Comment (WaVEV) [reply django_mongodb/features.py:70]: I think we should open a new issue for the first one. It's out of aggregation's scope.

Comment (timgraham) [reply django_mongodb/features.py]: Thanks. I squashed the third commit with the second since it was actually a regression in that commit.

**Review by timgraham — APPROVED** [review]

**Pull request merged**